### PR TITLE
fix: added check for empty path in 06-solomachine

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #1916 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/light-clients/06-solomachine/client_state.go [hunk 1]
```diff
@@ -125,6 +125,10 @@ func (cs *ClientState) VerifyMembership(
 		return sdkerrors.Wrap(commitmenttypes.ErrInvalidProof, "failed to unmarshal path into ICS 23 commitment merkle path")
 	}
 
+	if len(merklePath.String()) == 0 {
+		return sdkerrors.Wrap(commitmenttypes.ErrInvalidProof, "path is empty")
+	}
```
Comment (damiannolan) [inline modules/light-clients/06-solomachine/client_state.go:130]: Should we be using the `merklePath.Empty()` method which returns a `bool` instead?

File: modules/light-clients/06-solomachine/client_state_test.go [hunk 2]
```diff
@@ -541,6 +541,13 @@ func (suite *SoloMachineTestSuite) TestVerifyMembership() {
 				},
 				false,
 			},
+			{
+				"empty path",
+				func() {
+					path = []byte("")
+				},
+				false,
```
Comment (damiannolan) [inline modules/light-clients/06-solomachine/client_state_test.go:549]: I think this depends on https://github.com/cosmos/ibc-go/pull/2736 as `path` will change from `[]byte` to `exported.Path`

File: modules/light-clients/06-solomachine/client_state.go (hunk 1 above)
Comment (charleenfei) [reply modules/light-clients/06-solomachine/client_state.go:130]: ahaa, didnt realise that was a thing. thanks!

File: modules/light-clients/06-solomachine/client_state_test.go (hunk 2 above)
Comment (damiannolan) [reply modules/light-clients/06-solomachine/client_state_test.go:549]: #2736 is now merged, you may be able to change this to the below to cover the case:

```go
path = commitmenttypes.MerklePath{}
```

**Review by damiannolan — APPROVED** [review]
Nice, lgtm! 🚀

**Review by AdityaSripal — APPROVED** [review]

**Review by colin-axner — APPROVED** [review]
LGTM!

**Pull request merged**